scripts.py

```diff
@@ -1225,11 +1219,141 @@ def do_run(self):
+        editions.cover_id AS editions_cover_id, editions.cover_full_url AS editions_cover_full_url, editions.cover_thumbnail_url AS editions_cover_thumbnail_url, 
+        editions.open_access_download_url AS editions_open_access_download_url, editions.simple_opds_entry AS editions_simple_opds_entry, 
+        editions.extra AS editions_extra \n
+        FROM editions, identifiers \n
+        WHERE editions.primary_identifier_id IN (:primary_identifier_id_1) AND identifiers.type = :type_1'
+        '''
+
+        if log:
+            log.info(
+                "Processing %d editions.", query.count()
+            )
+        return query.order_by(Edition.id)
+
+
+    def run(self, batch_size=10):
+        param_args = self.parse_command_line(self._db)
+        
+        #if param_args.identifiers:
+            # we're asked about a specific set of work contributors
+            #identifier_ids = [x.id for x in param_args.identifiers]
+
+        self.query = self.make_query(
+            self._db, param_args.identifier_type, param_args.identifiers, self.log
+        )
+
+        editions = True
+        offset = 0
+        while editions:
+            editions = self.query.offset(offset).limit(batch_size).all()
+            for edition in editions:
+                if edition.contributions:
+                    for contribution in edition.contributions:
+                        self.process_contribution(self._db, contribution, edition)
+            offset += batch_size
+            self._db.commit()
+        self._db.commit()
+
+
+    @classmethod
+    def process_contribution(cls, _db, contribution, edition):
+        if not contribution or not edition:
+            return
+
+        contributor = contribution.contributor
+
+        identifier = edition.primary_identifier
+
+        computed_sort_name_local = display_name_to_sort_name(contributor.display_name, advanced=True)
+        success = cls.check_sort_name(computed_sort_name_local, contributor, edition, DataSource.INTERNAL_PROCESSING, CoverageRecord.REPAIR_SORT_NAME_OPERATION)
+        if not success:
+            output = "Contributor[%s]: contributor_sort_name=%s, contributor_display_name=%s: INTERNAL MISMATCH" % (contributor.id, contributor.sort_name, contributor.display_name)
+            print output.encode("utf8")
+
+
+    @classmethod
+    def check_sort_name(cls, computed_sort_name, contributor, edition, data_source, operation):
+        success = True
+        if computed_sort_name.strip().lower() != contributor.sort_name.strip().lower():
+            record, is_new = CoverageRecord.add_for(
+                edition=edition, data_source=DataSource.VIAF, operation=CoverageRecord.REPAIR_SORT_NAME_OPERATION,
+                status=CoverageRecord.TRANSIENT_FAILURE
+            )
+            # NOTE:  not the best way -- storing id info in the message field, but 
+            # perhaps better than making CoverageRecords cover Contributor-type objects.
```

Comment:
I split the code that deals with it into two portions.  The VIAF lookup code is in metadata, but the display_name --> sort_name code is in core.  So it is possible to resolve the transient failure from within core.  That code will follow in future push.

Not sure I understand the second part of the comment.